### PR TITLE
parse_qpcr_object

### PR DESCRIPTION
**Pull request by antgonza**

### CODE REVIEW

File: knimin/lib/parse.py [hunk 1]
```diff
@@ -34,3 +36,60 @@ def parse_plate_reader_output(contents):
...
+
+
+def parse_qpcr_object(contents):
+    """ Parses a QPCR machine string, returning the per well Cps
+
+    Parameters
+    ----------
+    contents : str
+        The string generated by the QPCR machine to parse.
+        New lines \n, field \t
```
Comment (josenavas) [inline knimin/lib/parse.py]: `\t` -> `,`

File: knimin/lib/parse.py [hunk 2]
```diff
@@ -34,3 +36,60 @@ def parse_plate_reader_output(contents):
...
+
+    # Pos format is A1 -> [0][0], B24 -> [1][23], etc so the max row will
+    # be the first letter - 'A' + 1
+    max_row = sorted(contents.Pos.values,
+                     key=lambda item: item[0])[-1][0]
+    max_row = ord(max_row) - ord('A') + 1
+    # the max columns is the max number after the letter - 1
+    max_col = sorted(contents.Pos.values,
+                     key=lambda item: int(item[1:]))[-1][1:]
+    max_col = int(max_col)
```
Comment (josenavas) [inline knimin/lib/parse.py]: isn't max col already an int?

File: knimin/lib/parse.py (hunk 1 above)
Comment (antgonza) [inline knimin/lib/parse.py]: the example you sent me has tabs not commas ...

File: knimin/lib/parse.py (hunk 2 above)
Comment (antgonza) [inline knimin/lib/parse.py]: nop, that int it's for sorting only - not passed back.

Comment (antgonza) [inline knimin/lib/parse.py]: turns out that google spreadsheets copy/pastes tabs even if the original file has commas ... changed in last pr.

File: knimin/lib/parse.py [hunk 3]
```diff
@@ -1,9 +1,43 @@
...
+
 from StringIO import StringIO
 
 import pandas as pd
 import numpy as np
 
 
+def parse_plate_reader_output(contents):
+    """Parses the output of a plate reader
+    The format supported here is a tab delimited file in which the first line
```
Comment (wasade) [inline knimin/lib/parse.py]: Space please

File: knimin/lib/tests/test_parse.py [hunk 4]
```diff
@@ -68,6 +68,95 @@ def test_parse_plate_reader_output(self):
...
+             7.77, 11.47, 15.52, 12.6, 11.18, 21.37, 15.07, 11.45, 18.32,
+             12.68, 11.52, 11.34, 24.32],
+            [6.61, 6.27, 12.11, 7.15, 6.21, 7.31, 6.87, 5.98, 6.43, 6.59, 6.5,
+             8.07, 7.23, 7.82, 10.09, 15.68, np.nan, np.nan, 14.78, 11.54,
+             np.nan, np.nan, np.nan, np.nan]]
+        npt.assert_almost_equal(obs, exp)
+
+        # errrors
+
+        with self.assertRaises(ValueError) as ctx:
```
Comment (wasade) [inline knimin/lib/tests/test_parse.py]: Possible for assertRaisesRegexp since there are multiple ways Value error can be thrown?

Comment (josenavas) [inline knimin/lib/tests/test_parse.py]: I think that is addressed in line 86?

Comment (wasade) [inline knimin/lib/tests/test_parse.py]: Oh I see, okay, +1

**Pull request merged**